Bio_Seq_Analysis_Tool.py

```diff
@@ -0,0 +1,163 @@
+    -gc_calculate - return sequence GC-content in percent
+    
+    :param args: nucleic acid sequence
+    :type param seqs: str
+    
+    :param operation: type of operation required
+    :type param operation: str
+    
+    :return: Analysis of nucleic acid sequence
+    :rtype: List[str]
+
+    :raises ValueError: if sequence not RNA or DNA, also if the operation value out of OPERATION_DICT
+    """
+    OPERATION_DICT = {'transcribe': na.transcribe, 
+                'reverse': na.reverse,
+                'reverse_complement': na.reverse_complement,
+                'complement': na.complement,
+                'gc_calculate': na.gc_calculate}
+    analysis = []
+    for seq in args:
+        na.is_na(seq)
+        if operation in OPERATION_DICT.keys():
+            analysis.append(OPERATION_DICT[operation](seq))
+        else:
+            raise ValueError(f'Wrong operation!')
+    return analysis
+
+
+def analyse_fastq(input_path: str, 
+                  gc_bounds: Union[int, float, Tuple [int], Tuple [float]] = (0, 100), 
+                  length_bounds: Union[int, Tuple [int]] = (0, 2**32),
+                  quality_threshold: float = 0.0, filtered_file_name: Union[None, str] = None) -> Dict[str,str]:
+    """
+    This function help analyze a set of reads obtained from next-generation sequencing. 
+    
+    The function allow to filter the desired reads according to three parameters:
+    GC-content, length and reading quality.
+    
+    :param seqs: 
+    Path to the file with FASTQ-sequences in the format. 
+    :type seqs: str
+    
+    :param gc_bounds: 
+    Boundary parameters for filtering sequences by GC-content. Save only reads with a GC-content between boundaries 
+    or lower than one boundary. Lower boundary cannot be less than 0 and upper boundary cannot be greater than 100. 
+    gc_bounds default value is (0,100).
+    :type param gc_bounds: Union[int, float, Tuple [int], Tuple [float]]
+    
+    :param length_bounds: 
+    Boundary parameters for filtering sequences by length. Works the same as gc_bounds. Lower boundary cannot be less 
+    than 0 and upper boundary cannot be greater than 2^32. length_bounds default value is (0,2^32)
+    :type param length_bounds: Union[int, Tuple [int]
+    
+    :param quality_threshold: 
+    Threshold for quality of each nucleotide in read. Quality incodes by ASCII codes. The threshold cannot be more 
+    than 40. quality_threshold default value is 0 
+    :type param quality_threshold: float
+    
+    :return: 
+    New dictionaries with fastq sequence.The first one consisting of filtered fastq sequences and the other one with 
+    sequences that did not pass filters.
+    :rtype: Dict[str]
+    
+    :raises ValueError: if sequence not RNA or DNA, also if the argument values are outside the allowed ones
+    """
+    seqs, path_to_file = fq.read_fastq(input_path)
+    file_name = path_to_file.split("/")[-1]
```

Comment:
Это ок, но мало ли у человека винда. Для этого есть вообще крутая штука:
```suggestion
    file_name = os.basename(path_to_file)
```